### ISSUE TITLE
Animated gif stops looped animation after resizing and bad quality

### ISSUE THREAD

**Issue by martonx** [SixLabors/ImageSharp#422]
Hi all,

I have an endless looped example gif file: 
![tictoc_balance_1024x1024](https://user-images.githubusercontent.com/407433/34470175-e14c1766-ef2b-11e7-98fe-3438292c24d8.gif)

When I resized this with code (asp.net core 2.0), I have two issues:

1. the endless loop stops after the first animation
2. the image quality is very bad

![tictoc_balance_500x500](https://user-images.githubusercontent.com/407433/34470213-e91916d2-ef2c-11e7-8cf0-ed06069e6a0b.gif)

Could you help me, what should I change on my code?

``` c#
public byte[] ResizeImage(byte[] imageByteArray, int? maxHeight = null, int? maxWidth = null, ImageFormat imageFormat = ImageFormat.Jpeg)
        {
            using (var original = new Image(imageByteArray))
            {
                int newWidth = original.Width;
                int newHeight = original.Height;

                if (maxWidth < original.Width || maxHeight < original.Height)
                {
                    var ratioX = (double)(maxWidth ?? original.Width) / original.Width;
                    var ratioY = (double)(maxHeight ?? original.Height) / original.Height;
                    var ratio = Math.Min(ratioX, ratioY);

                    newWidth = (int)(original.Width * ratio);
                    newHeight = (int)(original.Height * ratio);
                }

                using (var resized = original.Resize(newWidth, newHeight, new Lanczos2Resampler()))
                {
                    if (resized == null)
                        return null;

                    using (var ms = new MemoryStream())
                    {
                        switch (imageFormat)
                        {
                            case ImageFormat.Jpeg:
                                resized.SaveAsJpeg(ms);
                                break;
                            case ImageFormat.Png:
                                resized.SaveAsPng(ms);
                                break;
                            case ImageFormat.Gif:
                                resized.SaveAsGif(ms);
                                break;
                            default:
                                throw new NotSupportedException($"Image format {imageFormat} not supported!");
                        }

                        return ms.ToArray();
                    }
                }
            }
        }
```

**Comment by JimBobSquarePants** [comment]
Hi @martonx,

I've just tested both the image and the resampler using the latest beta on Nuget

https://www.nuget.org/packages/SixLabors.ImageSharp/1.0.0-beta0002

and I was able to resize the image without issue. 

![rings](https://user-images.githubusercontent.com/385879/34648955-d84d3be6-f3f8-11e7-92f5-23f16ba7cf14.gif)

Since you did not provide a version number as requested in the contribution guidelines when you opened the issue and that I can see you are using the `new Image(imageByteArray)` constructor I can only conclude that you are using the fake RevStackCore-ImageSharp that someone added to Nuget without our permission.

This a a copy of a very early alpha that should never be used. 

P.S

It looks like you are using `MemoryStream.ToArray()` a lot in your code. That's going to cause you issues down the line as you are allocating too much memory. I recommend using streams directly.
